# HG changeset patch
# Parent 0a9e4697d1d0fd94ea437831666aa475fa33a89b
Build: Prune python, leaving lowlevel bindings

UPSTREAM: Figure out why we need the lowlevel bindings, make a config option

diff -r 0a9e4697d1d0 tools/python/Makefile
--- a/tools/python/Makefile
+++ b/tools/python/Makefile
@@ -24,8 +24,7 @@ install: install-dtd
 		$(PYTHON_PREFIX_ARG) --root="$(DESTDIR)" --force
 
 install-dtd: all
-	$(INSTALL_DIR) $(DESTDIR)$(SHAREDIR)/xen
-	$(INSTALL_DATA) xen/xm/create.dtd $(DESTDIR)$(SHAREDIR)/xen
+	@ :
 
 .PHONY: test
 test:
diff -r 0a9e4697d1d0 tools/python/setup.py
--- a/tools/python/setup.py
+++ b/tools/python/setup.py
@@ -87,40 +87,14 @@ xl = Extension("xl",
                depends            = [ PATH_LIBXL + "/libxenlight.so" ],
                sources            = [ "xen/lowlevel/xl/xl.c", "xen/lowlevel/xl/_pyxl_types.c" ])
 
-plat = os.uname()[0]
-modules = [ xc, xs, ptsname, flask ]
-#modules.extend([ xl ])
-if plat == 'SunOS':
-    modules.extend([ scf, process ])
-if plat == 'Linux':
-    modules.extend([ checkpoint, netlink ])
+modules = [ xc, xs, ptsname ]
 
 setup(name            = 'xen',
       version         = '3.0',
       description     = 'Xen',
       packages        = ['xen',
-                         'xen.lowlevel',
-                         'xen.util',
-                         'xen.util.xsm',
-                         'xen.util.xsm.dummy',
-                         'xen.util.xsm.flask',
-                         'xen.util.xsm.acm',
-                         'xen.xend',
-                         'xen.xend.server',
-                         'xen.xend.xenstore',
-                         'xen.xm',
-                         'xen.web',
-                         'xen.sv',
-                         'xen.xsview',
-                         'xen.remus',
-                         'xen.xend.tests',
-                         'xen.xend.server.tests',
-                         'xen.xend.xenstore.tests',
-                         'xen.xm.tests'
+                         'xen.lowlevel'
                          ],
       ext_package = "xen.lowlevel",
       ext_modules = modules
       )
-
-os.chdir('logging')
-execfile('setup.py')
